### DISCUSSION TITLE
How to exclude empty types when using fragments?

### DISCUSSION THREAD

**Discussion by zo-ly** [dotansimha/graphql-code-generator#8812] [Q&A] [answered]
I am upgrading the codegen version in my project from 1.1.3 to 2 and have encountered a lot of typescript compatibility issues.

My problem is similar to this [discussion](https://github.com/dotansimha/graphql-code-generator/discussions/5567). I found that the v1.1.3 can exclude these empty types, but the v2 can't. Is there any way now to exclude these empty types?

Here is the [code demonstration](https://github.com/zo-ly/graphql-codegen-test). Thank you very much for this tool.

**Comment by dotansimha** [comment, marked as answer]
Hi @zo-ly ! 
I'm not sure what feature in v1 allowed to skip/exclude the "empty" types? 

I have to refer again to my comment here: https://github.com/dotansimha/graphql-code-generator/discussions/5567#discussioncomment-449391 
Those types are not empty - they represent a real, possible scenario that might happen during the execution of your GraphQL engine.  Ignoring these cases while querying might lead to runtime issues.
